### PR TITLE
feat: #625 ([A11Y] no-abstract-roles)

### PR DESCRIPTION
**Pull request by lifeart**

### CODE REVIEW

File: docs/rule/no-abstract-roles.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* structure
+* widget
+* window
+
+### `<* role>`
+
+This rule **forbids** the following:
+
+```hbs
+<button role="window">
```
Comment (MelSumner) [inline docs/rule/no-abstract-roles.md]: for consistency, let's use `<div role="window">`

File: docs/rule/no-abstract-roles.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+This rule **allows** the following:
+
+```hbs
+<img role="button">
```
Comment (MelSumner) [inline docs/rule/no-abstract-roles.md]: for consistency, let's use `<div role="button">`

File: lib/rules/lint-no-abstract-roles.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  visitor() {
+    return {
+      ElementNode(node) {
+        const roleAttr = AstNodeInfo.findAttribute(node, 'role');
+        if (roleAttr) {
+          const roleAttrValue = attributeTextValue(roleAttr);
+          if (PROHIBITED_ROLE_VALUES.includes(roleAttrValue)) {
+            this.logNode({
+              message:
+                'The HTML attribute role should never have the following value: ' + roleAttrValue,
```
Comment (MelSumner) [inline lib/rules/lint-no-abstract-roles.js]: roleAttrValue + ' is an abstract role, and is not a valid value for the role attribute. See https://www.w3.org/WAI/PF/aria/roles#abstract_roles for more information.'

File: test/unit/rules/lint-no-abstract-roles-test.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    '<img alt="" role="none" src="zoey.jpg">',
+    '<img alt="" role="presentation" src="zoey.jpg">',
+  ],
+
+  bad: [
+    {
+      template: '<img role="command">',
+
+      result: {
+        message: 'The HTML attribute role should never have the following value: command',
```
Comment (MelSumner) [inline test/unit/rules/lint-no-abstract-roles-test.js]: 'command is not a valid value for the role attribute.'

File: test/unit/rules/lint-no-abstract-roles-test.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        source: '<img role="command">',
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<img role="composite">',
+
+      result: {
+        message: 'The HTML attribute role should never have the following value: composite',
```
Comment (MelSumner) [inline test/unit/rules/lint-no-abstract-roles-test.js]: please refer to the above change request and update the others to match. thank you!

**Comment by lifeart** [comment]
@MelSumner done

**Pull request merged**